packages/compiler/src/init/init-template-validate.ts

```diff
@@ -0,0 +1,20 @@
+import { createJSONSchemaValidator } from "../core/schema-validator.js";
+import { Diagnostic, NoTarget, SourceFile } from "../index.js";
+import { InitTemplateSchema } from "./init-template.js";
+
+export type ValidationResult = {
+  valid: boolean;
+  diagnostics: readonly Diagnostic[];
+};
+
+export function validateTemplateDefinitions(
+  template: unknown,
+  templateName: SourceFile | typeof NoTarget,
+  strictValidation: boolean,
+): ValidationResult {
```

Comment:
Extract the validateTemplateDefinitions() method from init.ts into a separated file. otherwise, other projects like playground-website or html-program-view using vite will complain about "prompts" and "fs/promises" libraries which is included in the imports in init.ts.